Source/JavaScriptCore/parser/Lexer.cpp

```diff
@@ -752,7 +752,7 @@ static bool isNonLatin1IdentStart(char32_t c)
 template<typename CharacterType>
 static ALWAYS_INLINE bool isIdentStart(CharacterType c)
 {
-    static_assert(std::is_same_v<CharacterType, Latin1Character> || std::is_same_v<CharacterType, char32_t>, "Call isSingleCharacterIdentStart for UChars that don't need to check for surrogate pairs");
+    static_assert(std::is_same_v<CharacterType, Latin1Character> || std::is_same_v<CharacterType, char32_t>, "Call isSingleCharacterIdentStart instead");
```

Comment:
I changed the text here because `isSingleCharacterIdentStart` (and `isSingleCharacterIdentPart`) are both aware that surrogate pairs exist and return false if they encounter one. But that was a mistake; the functions just spuriously return false if they see a surrogate pair, rather than converting it to a codepoint as they probably ought to do.

The existing code is probably wrong too, though, because it's quite unusual to be certain that your UTF-16 input does not need to check for surrogate pairs. I think I should return to the original wording for these static asserts, and add FIXME comments instead.

Reply:
Good news is they are only called in a couple places. I don't understand the code, but the callers actually are checking for surrogates, so I'm going to assume it's OK and move on.

Maybe it would be better to name it `isSingleCodeUnitIdentStart` instead of `isSingleCharacterIdentStart`, but then again, probably that's not actually meaningful unless you're familiar with Unicode.

I'll reword the static assert message: "For char16_t, call isSingleCharacterIdentStart, but beware it does not handle surrogate pairs"